awscli/customizations/paginate.py

```diff
@@ -337,8 +338,16 @@ def add_to_parser(self, parser):
         parser.add_argument(self.cli_name, dest=self.py_name,
                             type=self.type_map[self._parse_type])
 
+    def _emit_non_positive_max_items_warning(self, value):
+        if int(value) <= 0:
+            uni_print(
+                "warning: Non-positive values for --max-items may result in undefined behavior.\n",
+                sys.stderr)
+
     def add_to_params(self, parameters, value):
         if value is not None:
+            if self._serialized_name == 'MaxItems':
```

Comment:
We should check `int(value) <=0` here and remove it from `_emit_non_positive_max_items_warning`.